pkg/scheduler/framework/plugins/noderesources/least_allocated.go

```diff
@@ -63,25 +64,45 @@ func (la *LeastAllocated) ScoreExtensions() framework.ScoreExtensions {
 }
 
 // NewLeastAllocated initializes a new plugin and returns it.
-func NewLeastAllocated(_ runtime.Object, h framework.FrameworkHandle) (framework.Plugin, error) {
+func NewLeastAllocated(laArgs runtime.Object, h framework.FrameworkHandle) (framework.Plugin, error) {
+	args, err := getNodeResourcesLeastAllocatedArgs(laArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	resToWeightMap := make(resourceToWeightMap)
+	for _, resource := range args.Resources {
+		resToWeightMap[v1.ResourceName(resource.Name)] = resource.Weight
+	}
+
 	return &LeastAllocated{
 		handle: h,
 		resourceAllocationScorer: resourceAllocationScorer{
-			LeastAllocatedName,
-			leastResourceScorer,
-			defaultRequestedRatioResources,
+			Name:                LeastAllocatedName,
+			scorer:              leastResourceScorer(resToWeightMap),
+			resourceToWeightMap: resToWeightMap,
 		},
 	}, nil
 }
 
-func leastResourceScorer(requested, allocable resourceToValueMap, includeVolumes bool, requestedVolumes int, allocatableVolumes int) int64 {
-	var nodeScore, weightSum int64
-	for resource, weight := range defaultRequestedRatioResources {
-		resourceScore := leastRequestedScore(requested[resource], allocable[resource])
-		nodeScore += resourceScore * weight
-		weightSum += weight
+func getNodeResourcesLeastAllocatedArgs(obj runtime.Object) (config.NodeResourcesLeastAllocatedArgs, error) {
+	ptr, ok := obj.(*config.NodeResourcesLeastAllocatedArgs)
+	if !ok {
+		return config.NodeResourcesLeastAllocatedArgs{}, fmt.Errorf("want args to be of type NodeResourcesLeastAllocatedArgs, got %T", obj)
+	}
+	return *ptr, nil
+}
+
+func leastResourceScorer(resToWeightMap resourceToWeightMap) func(resourceToValueMap, resourceToValueMap, bool, int, int) int64 {
+	return func(requested, allocable resourceToValueMap, includeVolumes bool, requestedVolumes int, allocatableVolumes int) int64 {
+		var nodeScore, weightSum int64
+		for resource, weight := range resToWeightMap {
+			resourceScore := leastRequestedScore(requested[resource], allocable[resource])
+			nodeScore += resourceScore * weight
+			weightSum += weight
+		}
+		return nodeScore / weightSum
```

Comment:
one user could put all weights to zero. Let's return a zero score in that case, otherwise this is a division by zero.

Reply:
This won't happen again when the zero weight has been skipped.

Reply:
the sum could still be zero. But, as I said in the other comment, we should disallow 0 instead.

Reply:
this is not resolved unless you error on 0